### DISCUSSION TITLE
Storing user first/last name, other data

### DISCUSSION THREAD

**Discussion by mariusa** [supabase/supabase#226] [Questions] [answered]
Hi,

I see user first/last/full name isn't stored in 
https://github.com/supabase/gotrue-js/blob/05e6c25056fc310d1b6e5c6097bfaa4825dce536/infra/db/00-schema.sql

How do you recommend to store this & other user data?

**Comment by kiwicopple** [comment, marked as answer]
There are a few ways to do this. I recommend that you keep all the user data in a completely separate table. You will have: 

- `auth.users` - GoTrue only, session data
- `public.users` - For your app. Will contain the user data. I recommend setting up the table something like this:

```sql
create table public.users (
  id            uuid primary key references auth.users,
  first_name    text,
  last_name     text
);
```

You can also create a trigger on the `auth.users` table that inserts a row into the public.users table, but it's not necessary - you can handle this from your application so that once they have successfully signed up, you are returned a `uuid`, and you insert data into `public.users`.

**Reply by mariusa** [reply]
Thanks!

**Reply by imjared** [reply]
@kiwicopple - when you mention "trigger" here, are you talking about a workflow that happens on the client? ideally i'd like for this to be happening on a server somewhere but https://github.com/supabase/supabase/discussions/259 makes me think we're not quite there yet and i must contain my excitement :)

**Reply by kiwicopple** [reply]
> ideally i'd like for this to be happening on a server

Yes it happens in the database :). The database is just Postgres, and Postgres has [triggers](https://www.postgresql.org/docs/current/sql-createtrigger.html). 

Here is an example that you could run in the SQL editor on the dashboard: https://github.com/supabase/supabase/discussions/306#discussioncomment-138424

**Reply by lior-amsalem** [reply]
for safety you should enable row level security 
`alter table public.profiles enable row level security;` 
so no sensitive data will be acces to public via anon requst

**Comment by pankaj-wwtl** [comment]
@kiwicopple 
What table structure would you recommend if two types of users can log in to the system? like teachers and students.
